translate/storage/pypo.py

```diff
@@ -113,127 +113,268 @@ def unicode_width(text: str) -> int:
-            )
-        """,
-        re.VERBOSE,
-    )
-
     def __init__(self, width: int = 77) -> None:
-        super().__init__(
-            width=width,
-            replace_whitespace=False,
-            expand_tabs=False,
-            drop_whitespace=False,
-            break_long_words=True,
-        )
+        self.width = width
 
     def __eq__(self, other) -> bool:
         if not isinstance(other, PoWrapper):
             return False
         return self.width == other.width
 
-    def _handle_long_word(
-        self, reversed_chunks: list[str], cur_line: list[str], cur_len: int, width: int
-    ) -> None:
+    def wrap(self, text: str) -> list[str]:
         """
-        Handle a chunk of text (most likely a word, not whitespace) that
-        is too long to fit in any line.
+        Wrap text to fit within the specified width.
+
+        Returns a list of lines, each no longer than self.width characters
+        (accounting for character display width for CJK characters).
         """
-        # We're allowed to break long words, then do so: put as much
-        # of the next chunk onto the current line as will fit.
-        chunk_start, chunk_end = cjkslices(reversed_chunks[-1], width - cur_len)
-        cur_line.append(chunk_start)
-        reversed_chunks[-1] = chunk_end
+        if not text or self.width <= 0:
+            return [text] if text else []
+
+        # Get display width of text
+        text_width = unicode_width(text)
+
+        # If text fits on one line, return it as-is
+        if text_width <= self.width:
+            return [text]
+
+        # Split text into chunks using word separator pattern
+        chunks = self._split_chunks(text)
+
+        # Now wrap the chunks into lines
+        return self._wrap_chunks(chunks)
+
+    def _split_chunks(self, text: str) -> list[str]:
+        """
+        Split text into chunks at word boundaries using Unicode line breaking (UAX#14).
+
+        Uses the uniseg library's line_break_units() which implements the
+        Unicode Line Breaking Algorithm. Post-processes to handle:
+        - Escape sequences as atomic units
+        - Slash-separated paths for URLs
+        """
+        try:
+            # Get line break units using UAX#14
+            units = list(line_break_units(text))
+            # Post-process units to handle escape sequences and slashes better
+            processed_units = []
+            for unit in units:
+                # If unit contains escape sequences or slashes, further split it
+                if "\\\\" in unit or ("/" in unit and "[" not in unit):
+                    # Use manual chunking for this unit
+                    sub_chunks = self._manual_chunk(unit)
+                    processed_units.extend(sub_chunks)
+                elif unit:  # Skip empty units
+                    processed_units.append(unit)
+            if processed_units:
+                return processed_units
+        except Exception:
```

Comment:
Remove this exception handling, we should IMHO fail if there is an error and not hide it here.